pkg/util/cache/intern.go

```diff
@@ -0,0 +1,359 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package cache
+
+import (
+	"fmt"
+	cconfig "github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	lru "github.com/hashicorp/golang-lru"
+	"go.uber.org/atomic"
+	"math"
+	"os"
+	"sync"
+	"time"
+	"unsafe"
+)
+
+// initialInternerSize is the size of the LRU cache (in #strings).  This is HEAP, so
+// don't let this get too big compared to the MMAP region.
+const initialInternerSize = 64
+
+// backingBytesPerInCoreEntry is the number of bytes to allocate in the mmap file per
+// element in our LRU.  E.g., some value of initialInternerSize * POW(growthFactor, N).
+const backingBytesPerInCoreEntry = 4096
+const growthFactor = 1.5
+
+// noFileCache indicates that no mmap should be created.
+const noFileCache = ""
+
+// OriginTimeSampler marks allocations to the Time Sampler.
+const OriginTimeSampler = "!Timesampler"
```

Comment:
Should this really be defined here ?

Reply:
These are for internal diagnostic use.  I don't care where they go, but it's easier to see the full list if they're all in one place.  I can move them to their use sites if preferred.